pkg/ipamd/ipamd.go

```diff
@@ -1771,10 +1776,12 @@ func (c *IPAMContext) filterUnmanagedENIs(enis []awsutils.ENIMetadata) []awsutil
 		} else if c.awsClient.IsCNIUnmanagedENI(eni.ENIID) {
 			log.Debugf("Skipping ENI %s: since on non-zero network card", eni.ENIID)
 			numFiltered++
+			nonZeroCard++
```

Comment:
Discussed offline, and while we still need to filter for ENIs behind the non-zero network card, there is no need to count the number here since we already have that number in the result of the describe call, i.e. `metadataResult.MultiCardENIIDs`

Reply:
We also want to rename `IsCNIUnmanagedENI` to something that clearly indicates that we are checking if this is an ENI that is not behind Network Card 0, i.e. a multi-card ENI